diff --git a/base/Dockerfile b/base/Dockerfile
index cca3b3e..bda3eff 100644
--- a/base/Dockerfile
+++ b/base/Dockerfile
@@ -1,5 +1,5 @@
 # This image is the base image for all OpenShift v3 language container images.
-FROM centos/s2i-core-centos7
+FROM localhost/clefos/s2i-core-centos7
 
 ENV SUMMARY="Base image with essential libraries and tools used as a base for \
 builder images like perl, python, ruby, etc." \
@@ -14,7 +14,7 @@ LABEL summary="$SUMMARY" \
       io.k8s.description="$DESCRIPTION" \
       io.k8s.display-name="s2i base" \
       com.redhat.component="s2i-base-container" \
-      name="centos/s2i-base-centos7" \
+      name="clefos/s2i-base-centos7" \
       version="1" \
       release="1" \
       maintainer="SoftwareCollections.org <sclorg@redhat.com>"
@@ -46,6 +46,6 @@ RUN yum install -y centos-release-scl && \
   wget \
   which \
   zlib-devel" && \
-  yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
+  yum install -y --setopt=tsflags=nodocs --disablerepo=epel $INSTALL_PKGS && \
   rpm -V $INSTALL_PKGS && \
   yum clean all -y
diff --git a/core/Dockerfile b/core/Dockerfile
index 620c6f5..5e96322 100644
--- a/core/Dockerfile
+++ b/core/Dockerfile
@@ -13,7 +13,7 @@ LABEL summary="$SUMMARY" \
       io.openshift.s2i.scripts-url=image:///usr/libexec/s2i \
       io.s2i.scripts-url=image:///usr/libexec/s2i \
       com.redhat.component="s2i-core-container" \
-      name="centos/s2i-core-centos7" \
+      name="clefos/s2i-core-centos7" \
       version="1" \
       release="1" \
       maintainer="SoftwareCollections.org <sclorg@redhat.com>"
@@ -41,7 +41,7 @@ ENV BASH_ENV=${APP_ROOT}/etc/scl_enable \
 # Also setup the 'openshift' user that is used for the build execution and for the
 # application runtime execution.
 # TODO: Use better UID and GID values
-RUN rpmkeys --import file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7 && \
+RUN rpmkeys --import file:///etc/pki/rpm-gpg/RPM-GPG-KEY-ClefOS-7 && \
   INSTALL_PKGS="bsdtar \
   findutils \
   gettext \
@@ -52,7 +52,7 @@ RUN rpmkeys --import file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7 && \
   yum-utils" && \
   mkdir -p ${HOME}/.pki/nssdb && \
   chown -R 1001:0 ${HOME}/.pki && \
-  yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
+  yum install -y --setopt=tsflags=nodocs --disablerepo=epel $INSTALL_PKGS && \
   rpm -V $INSTALL_PKGS && \
   yum clean all -y
 
