diff --git a/3.11/Dockerfile b/3.11/Dockerfile
index acb85d2..f28a86c 100644
--- a/3.11/Dockerfile
+++ b/3.11/Dockerfile
@@ -1,6 +1,6 @@
-FROM centos/s2i-core-centos7
+FROM clefos/cassandra-container
 
-# Cassandra image for CentOS.
+# Cassandra image for ClefOS.
 #
 # Volumes:
 #  * /var/opt/rh/sclo-cassandra3/lib/cassandra - Datastore for Cassandra
@@ -31,7 +31,7 @@ LABEL summary="$SUMMARY" \
       io.k8s.display-name="Cassandra $CASSANDRA_VERSION" \
       io.openshift.expose-services="7199:cassandra,9042:cql" \
       io.openshift.tags="database,cassandra,cassandra3" \
-      name="centos/cassandra-311-centos7" \
+      name="clefos/cassandra-311-clefos" \
       version="$CASSANDRA_VERSION" \
       com.redhat.component="rh-cassandra3-docker" \
       maintainer="SoftwareCollections.org <sclorg@redhat.com>"
@@ -43,8 +43,8 @@ LABEL summary="$SUMMARY" \
 EXPOSE 7000 7001 7199 9042
 
 RUN INSTALL_PKGS="sclo-cassandra3 sclo-cassandra3-cassandra jemalloc" && \
-    yum install -y centos-release-scl epel-release && \
-    yum install --enablerepo=centos-sclo-sclo-testing -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
+    yum install -y clefos-release-scl epel-release && \
+    yum install --enablerepo=clefos-sclo-sclo-testing -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
     yum clean all
